From f2d69f02330265578f603b0d6866753b80948073 Mon Sep 17 00:00:00 2001
From: neon <srdjanneon@gmail.com>
Date: Sun, 14 Mar 2021 02:41:38 +0100
Subject: [PATCH] Disable startup chunk loading


diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index e779442b3..fb94f9f5a 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -350,7 +350,7 @@ public abstract class MinecraftServer implements Runnable, ICommandListener, IAs
             long j = az();
             i = 0;
 
-            for (int k = -192; k <= 192 && this.isRunning(); k += 16) {
+            /*for (int k = -192; k <= 192 && this.isRunning(); k += 16) {
                 for (int l = -192; l <= 192 && this.isRunning(); l += 16) {
                     long i1 = az();
 
@@ -362,7 +362,7 @@ public abstract class MinecraftServer implements Runnable, ICommandListener, IAs
                     ++i;
                     worldserver.chunkProviderServer.getChunkAt(blockposition.getX() + k >> 4, blockposition.getZ() + l >> 4);
                 }
-            }
+            }*/
         }
 
         for (WorldServer world : this.worlds) {
-- 
2.30.2

